### ISSUE TITLE
False reporting with @typescript-eslint/no-empty-interface when using Generics

### ISSUE THREAD

**Issue by moltar** [typescript-eslint/typescript-eslint#351]
## Rule

`@typescript-eslint/no-empty-interface`

## Problem Description

I am inheriting from a generic interface to create a specific one for each status type. Not all of these need to override or add any properties. E.g. a failure one might add `error` but a matching success one adds/overrides nothing, yet it still needs to be defined, as a generic interface cannot be used on it's own.

## Code Sample

```
const enum Status {
  SUCCESS,
  FAILURE,
}

interface GenericResponse<Status> {
  status: Status
}

interface SuccessResponse extends GenericResponse<Status.SUCCESS> {}
```

## Screenshot

<img width="515" alt="image" src="https://user-images.githubusercontent.com/491247/54250939-3a6b1880-4578-11e9-828a-9a8b4f2f5f92.png">

**Expected Result**

No warnings.

**Actual Result**

Has warnings.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.2` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `5.15.1` |
| `node`                             | `v11.6.0` |
| `npm`                              | `6.9.0` |

**Comment by ThomasdenH** [comment]
What do you mean by "a generic interface cannot be used on its own?" The code below is valid Typescript, where extending the interface is unnecessary.

```Typescript
const enum Status {
  SUCCESS,
  FAILURE,
}

interface GenericResponse<Status> {
  status: Status
}

function f(s: GenericResponse<Status.SUCCESS>): void {
    console.log(f);
}

const a: GenericResponse<Status.SUCCESS> = { status: Status.SUCCESS };

f(a);
```

**Comment by moltar** [comment]
I mean it cannot be just extended, without adding any properties. The rule gets triggered. I don't think it is valid. Sure there are no props added, but the generic is turned into specific by providing interface casts or whatever it is called.

**Comment by ThomasdenH** [comment]
I'd say that's an argument against the rule, not a bug with it. You could also use a type definition:

```Typescript
type SuccessResponse = GenericResponse<Status.SUCCESS>;
```

**Comment by moltar** [comment]
Oh, right! That works for me. Didn't think of that. Thanks! Issue closed :)
